src/components/DxpFacilitySwitcher.vue

```diff
@@ -0,0 +1,160 @@
+            <ion-icon :icon="closeOutline"/>
+          </ion-button>
+        </ion-buttons>
+        <ion-title>{{ $t("Select Facility") }}</ion-title>
+      </ion-toolbar>
+    </ion-header>
+    <ion-toolbar>
+      <ion-searchbar @ionFocus="selectSearchBarText($event)" :placeholder="$t('Search facilities')" v-model="queryString" @keyup.enter="queryString = $event.target.value; findFacility()" @keydown="preventSpecialCharacters($event)"/>
+    </ion-toolbar>
+    <ion-content>
+      <ion-radio-group v-model="selectedFacilityId">
+        <ion-list>
+          <!-- Loading state -->
+          <div class="empty-state" v-if="isLoading">
+            <ion-item lines="none">
+              <ion-spinner color="secondary" name="crescent" slot="start" />
+              {{ $t("Fetching facilities") }}
+            </ion-item>
+          </div>
+          <!-- Empty state -->
+          <div class="empty-state" v-else-if="filteredFacilities.length === 0">
+            <p>{{ $t("No facilities found") }}</p>
+          </div>
+          <div v-else>
+            <ion-item v-for="facility in filteredFacilities" :key="facility.facilityId">
+              <ion-radio label-placement="end" justify="start" :value="facility.facilityId">
+                <ion-label>
+                  {{ facility.facilityName }}
+                  <p>{{ facility.facilityId }}</p>
+                </ion-label>
+              </ion-radio>
+            </ion-item>
+          </div>
+        </ion-list>
+      </ion-radio-group>
+
+      <ion-fab vertical="bottom" horizontal="end" slot="fixed">
+        <ion-fab-button :disabled="selectedFacilityId === currentFacility.facilityId" @click="setFacility">
+          <ion-icon :icon="saveOutline" />
+        </ion-fab-button>
+      </ion-fab>
+    </ion-content>
+  </ion-modal>
+</template>
+
+<script setup lang="ts">
+import { 
+  IonButton,
+  IonButtons,
+  IonCard,
+  IonCardContent,
+  IonCardHeader,
+  IonCardTitle,
+  IonContent,
+  IonFab,
+  IonFabButton,
+  IonHeader,
+  IonIcon,
+  IonItem,
+  IonLabel,
+  IonList,
+  IonModal,
+  IonRadio,
+  IonRadioGroup,
+  IonSearchbar,
+  IonSpinner,
+  IonTitle,
+  IonToolbar
+} from '@ionic/vue';
+import { closeOutline, saveOutline } from "ionicons/icons";
+import { useUserStore } from 'src/store/user';
+import { computed, ref } from 'vue';
+
+const userStore = useUserStore();
+
+const facilities = computed(() => userStore.getFacilites)
+const currentFacility = computed(() => userStore.getCurrentFacility)
+
+const facilityModal = ref()
+const queryString = ref('')
+const isLoading = ref(true);
+const filteredFacilities = ref([])
+const selectedFacilityId = ref(currentFacility.value.facilityId)
+
+const emit = defineEmits(["updateFacility"])
+
+const closeModal = () => {
+  facilityModal.value.$el.dismiss(null, 'cancel');
+}
+
+function loadFacilities() {
+  filteredFacilities.value = facilities.value;
+  isLoading.value = false;
+}
+
+const findFacility = () => {
```

Comment:
Add check before searching to check if the value is empty, also when matching the value trim the search string.